src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let arrayInString = '';
+
+    if (this === undefined || this === null || this.length === 0) {
+      return '';
+    }
```

Comment:
This is unnecessary check, you could remove this code. Because this way only line 24 will execute even without check.